specs/014-kratos-authid/spec.md

```diff
@@ -0,0 +1,132 @@
+
+**Why this priority**: Without the linkage, support teams cannot validate identities for the current user base, creating immediate operational risk.
+
+**Independent Test**: Run the migration on a staging snapshot and verify each existing account lists a Kratos identity identifier retrieved from Kratos.
+
+**Acceptance Scenarios**:
+
+1. **Given** a populated user database without stored Kratos identifiers, **When** the migration executes, **Then** each active user record persists the correct Kratos identity identifier sourced from Kratos.
+2. **Given** an existing user whose Kratos identity cannot be resolved automatically, **When** the migration completes, **Then** the user is flagged for follow-up with a clear audit log entry explaining the missing identifier.
+
+---
+
+### User Story 3 - Capture identity for newly created users (Priority: P3)
+
+As an onboarding flow owner, I need each newly created Alkemio user to automatically carry its Kratos identity identifier so future authentications and audits stay consistent.
+
+**Why this priority**: It prevents new accounts from introducing gaps in identity tracking immediately after the migration.
+
+**Independent Test**: Create a new user via the standard registration flow and confirm the Kratos identifier is stored alongside the user record without manual intervention.
+
+**Acceptance Scenarios**:
+
+1. **Given** a successful Kratos registration that produces a new identity, **When** the user account is created in Alkemio, **Then** the matching Kratos identity identifier is stored with the user record before the onboarding flow completes.
+2. **Given** a registration attempt where storing the Kratos identifier fails, **When** the system detects the failure, **Then** it blocks completion of the user creation and returns a clear error to the onboarding flow for retry or escalation.
+
+---
+
+### User Story 4 - Provide traceability for support investigations (Priority: P4)
+
+Support specialists need to quickly retrieve and confirm a user's Kratos identity identifier when responding to access or compliance inquiries.
+
+**Why this priority**: Fast traceability shortens incident resolution time and ensures regulatory reporting accuracy.
+
+**Independent Test**: Using administrative tooling, search for a user account and confirm the Kratos identity identifier is visible and auditable within one query.
+
+**Acceptance Scenarios**:
+
+1. **Given** a support operator viewing a user profile, **When** they inspect identity details, **Then** the stored Kratos identifier is displayed or exportable for verification purposes.
+2. **Given** a support audit export, **When** the export runs, **Then** it includes the Kratos identifier for each user so cross-system reconciliations succeed on the first attempt.
+
+---
+
+### Edge Cases
+
+- How does the migration handle users whose Kratos identity no longer exists or cannot be retrieved at runtime?
```

Comment:
That's possible currently, and if the user registers again, they will have their Alkemio profile. With the new approach matching by `authId` --> `kratosId` I don't see how that behavior will be preserved.